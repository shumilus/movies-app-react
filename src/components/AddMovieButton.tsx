import { createUseStyles } from 'react-jss';

interface AddMovieButtonProps {
  title: string;
}

const useStyles = createUseStyles({
  button: {
    borderRadius: '4px',
    fontWeight: '600',
    fontSize: '20px',
    lineHeight: '24px',
    textTransform: 'uppercase',
    padding: '11px 18px',
    boxSizing: 'border-box',
    border: 'none',
    cursor: 'pointer',
  },
});

function AddMovieButton({ title }: AddMovieButtonProps) {
  const classes = `${useStyles().button} transparent-button`;

  return (
    <button type='button' className={classes}>{title}</button>
  );
}

export default AddMovieButton;
