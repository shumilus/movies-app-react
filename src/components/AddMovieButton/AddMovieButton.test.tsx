import { screen, render } from '@testing-library/react';
import AddMovieButton from './AddMovieButton';
import userEvent from '@testing-library/user-event';

describe('AddMovieButton', () => {
  it('should match AddMovieButton snapshot', () => {
    const { asFragment } = render(<AddMovieButton title='add movie' handleClickAdd={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should trigger onSearchClick on button click', () => {
    const handleClick = jest.fn();

    render(<AddMovieButton title='add button' handleClickAdd={handleClick} />);
    userEvent.click(screen.getByRole('button'));

    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
