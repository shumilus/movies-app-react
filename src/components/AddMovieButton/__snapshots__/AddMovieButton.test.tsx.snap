// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovieButton should match AddMovieButton snapshot 1`] = `
<DocumentFragment>
  <button
    class="button-0-2-1 transparent-button"
    type="button"
  >
    add movie
  </button>
</DocumentFragment>
`;
